sdk/monitor/azure-monitor-opentelemetry-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/AzureMonitorInstaller.java

```diff
@@ -0,0 +1,113 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.monitor.opentelemetry.exporter;
+
+import com.azure.monitor.opentelemetry.exporter.implementation.AzureMonitorExporterProviderKeys;
+import com.azure.monitor.opentelemetry.exporter.implementation.AzureMonitorLogRecordExporterProvider;
+import com.azure.monitor.opentelemetry.exporter.implementation.AzureMonitorMetricExporterProvider;
+import com.azure.monitor.opentelemetry.exporter.implementation.AzureMonitorSpanExporterProvider;
+import io.opentelemetry.sdk.autoconfigure.AutoConfiguredOpenTelemetrySdkBuilder;
+import io.opentelemetry.sdk.metrics.Aggregation;
+import io.opentelemetry.sdk.metrics.InstrumentSelector;
+import io.opentelemetry.sdk.metrics.View;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Consumer;
+
+/**
+ * Use {@link AutoConfiguredOpenTelemetrySdkBuilder} to install the azure monitor exporters.
+ */
+public final class AzureMonitorInstaller {
+
+    private final AzureMonitorExporterBuilder azureMonitorExporterBuilder = new AzureMonitorExporterBuilder();
+
+    /**
+     * Construct an instance of {@link AzureMonitorInstaller}
+     */
+    AzureMonitorInstaller() {
+    }
+
+    /**
+     * Sets the connection string to use for exporting telemetry events to Azure Monitor.
+     *
+     * @param connectionString The connection string for the Azure Monitor resource.
+     * @return the updated {@link AzureMonitorInstaller} object.
+     */
+    public AzureMonitorInstaller connectionString(String connectionString) {
+        azureMonitorExporterBuilder.connectionString(connectionString);
+        return this;
+    }
+
+    /**
+     * Customize the {@link AzureMonitorExporterBuilder} which is used to during {@link #install(AutoConfiguredOpenTelemetrySdkBuilder)}.
+     * @param exporterCustomizer the exporterCustomizer value to set.
+     * @return the updated {@link AzureMonitorInstaller} object.
+     */
+    public AzureMonitorInstaller exporterCustomizer(Consumer<AzureMonitorExporterBuilder> exporterCustomizer) {
+        exporterCustomizer.accept(azureMonitorExporterBuilder);
+        return this;
+    }
+
+    /**
+     * Configures an {@link AutoConfiguredOpenTelemetrySdkBuilder} based on the options set in the builder.
+     * @param autoConfiguredOpenTelemetrySdkBuilder the {@link AutoConfiguredOpenTelemetrySdkBuilder} object.
+     */
+    public void install(AutoConfiguredOpenTelemetrySdkBuilder autoConfiguredOpenTelemetrySdkBuilder) {
```

Comment:
this violates ExternalDependencyExposedCheck lint check.

lint message: "Class   'AutoConfiguredOpenTelemetrySdkBuilder', is a class from external dependency.   You should not use it as a method argument type."



















































l

Reply:
we can suppress this check, see https://github.com/Azure/azure-sdk-for-java/blob/0f379118d823884583bd43f1aa4f5e6ac9decab9/eng/code-quality-reports/src/main/resources/checkstyle/checkstyle-suppressions.xml#L231-L232